dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -124,8 +124,11 @@ public T createHosted(@ApiParam(hidden = true) @Auth User user,
 
         // check if the user has hit a limit yet
         final long currentCount = getEntryDAO().countAllHosted(user.getId());
-        if (currentCount >= calculatedEntryLimit) {
-            throw new CustomWebApplicationException("You have " + currentCount + " workflows which is at the current limit of " + calculatedEntryLimit, HttpStatus.SC_PAYMENT_REQUIRED);
+        // The user parameter is cached via the AuthenticationCachePolicy and may not reflect the latest changes to the user
+        User updatedUser = userDAO.findById(user.getId());
```

Comment:
Hmmm, might be more efficient to just invalidate the cache when a user's limits are changed rather than checking each time. https://www.dropwizard.io/0.8.1/dropwizard-auth/apidocs/io/dropwizard/auth/CachingAuthenticator.html#invalidate-C-